manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/BridgeWorker.java

```diff
@@ -200,4 +213,29 @@ protected boolean isDeprovisioningComplete(Bridge managedResource) {
         //As far as the Worker mechanism is concerned work for a Bridge is complete when the dependencies are complete.
         return DEPROVISIONING_COMPLETED.contains(managedResource.getDependencyStatus());
     }
+
+    @Override
+    protected void recordError(Work work, Exception e) {
+        String bridgeId = work.getManagedResourceId();
+        Bridge bridge = getDao().findById(bridgeId);
+        BridgeErrorInstance bridgeErrorInstance = bridgeErrorHelper.getBridgeErrorInstance(e);
+        bridge.setBridgeErrorId(bridgeErrorInstance.getId());
+        bridge.setBridgeErrorUUID(bridgeErrorInstance.getUUID());
+        persist(bridge);
+    }
+
+    // The ErrorHandler Processor is a special type. It is not visible to Users and hence
+    // should it fail to be provisioned (or de-provisioned) the User is unaware of the reason.
+    // Therefore, propagate the error associated with the Error Handler Processor to the Bridge.
+    protected void propagateProcessorError(Bridge bridge) {
```

Comment:
@manstis If we think about what originally started this discussion e.g. "user doesn't know that their error handler has failed to provision", do we feel this is actually helping?

Is it easier just to have a `bridge.setStatusMessage("Failed to provision the Error Handler");`?

Reply:
Yes, I do believe this is helping (propagate the `Processor` failure to the `Bridge`).

I propagate the error code whereas you're proposing a generic message should the `Processor` be `FAILED`. 

I am not keen on storing or propagating messages as I believe error codes better support i18n.

Reply:
@manstis To be a bit clearer on what I meant. If the error propagated to the `Bridge` is something like "Failed to provision a ReplicaSet", how does the user know whether that relates to their Error Handler or the Bridge?

Is it better for us to have a ErrorHandler-specific code so that if the `FAILURE` status for a bridge is caused by the error handler, its much clearer for the user as to which component is actually causing the failure?

Reply:
@manstis Gentle nudge for your opinion on this one.